PB-393: Add search facets.

jkaeser opened the pull request:
https://jira.zivtech.com/browse/PB-393

jkaeser committed:
Giving each recipe its own page.

jkaeser committed:
Adding category filters.

jkaeser committed:
Indexing additional DOM elements.

jkaeser committed:
Readding recipe UIDs.

These are used to target recipes for rendering in docs pages.
This lets us write the recipe once and print it where needed.

jkaeser committed:
Programmatically applying facet filters.

jkaeser committed:
Moving no search text to script.

This message results from user interaction. It makes sense to let
it be controlled by the javascript.

jkaeser committed:
Remembering previous search paramaters.

jkaeser committed:
Writing function documentation.

jkaeser committed:
Adding form reset button.

jkaeser committed:
Placing Algolia logo.

We are on the free plan and are required to do this.

jkaeser committed:
Making filter options deselectable.

jkaeser committed:
Moving search scripts to separate file.

This allows us to load the search js only on the search page.

jkaeser committed:
Running search on form update.

jkaeser committed:
Initializing form values based on URL params.

jkaeser committed:
Ensuring search updates when a form element changes.

tortillaj commented on _includes/search-results.html:
```diff
@@ -1,3 +1,29 @@
+<div class="search__algolia">
+  <p>Search powered by
+    <a href="https://www.algolia.com/">
+      <?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Let's try to put the SVG in an include so we can run [gulp-svgmin](https://github.com/ben-eb/gulp-svgmin) against it.

tortillaj commented:
Going to quibble with the layout a bit. Let's try to get them on one line, like this:

![img_1432](https://user-images.githubusercontent.com/851788/27394249-827f437a-5679-11e7-81b6-80baff668fa2.jpg)

When we add more facets, we'll just reduce the size of the text input to support another one inline ...

p.s. submit inputs and buttons don't need labels because their value describes them well enough for a screen reader.

We can look at the JS together during probo day!

jkaeser committed:
Moving Algolia logo to include.

This will allow us to use gulp-svgmin.

jkaeser committed:
Adjusting search form layout.

jkaeser committed:
Updating query when filter value is selected.

jkaeser committed:
Tidying up behavioral loose ends.

1. Ensuring pressing return key updates search and does not
reload page.
2. Ensuring proper messages are set based on current form state.
3. Ensuring reset clears all form values and URL values;

jkaeser committed:
Making small style tweaks.

jkaeser committed:
Fixing Algolia logo link extending beyond image.

jkaeser committed:
Object prototype-ifying the search js.

jkaeser committed:
Caching search results.

tortillaj commented:
![terrific](https://user-images.githubusercontent.com/851788/27498461-1980bbe6-582d-11e7-8825-7bb08b0c6cd0.gif)

jkaeser committed:
Excluding pages from search results.

It is more confusing than helpful for the search page to be a
possible result. The file this commit removes was never publicly
exposed - we should write a new version if we want to provide
such information.
